fitbit-react/src/Components/VerticalCard/VerticalCard.js

```diff
@@ -49,7 +60,29 @@ function VerticalCard({ product }) {
         <div className="text-container">
           <h5 className="font-medium-large weight-semi-bold primary-text-color card-vertical-heading">
             {productTitle}
-            <AiFillHeart style={{ fontSize: "1.7rem" }} />
+            <AiFillHeart
+              style={
+                ifProductExistsInWishList
+                  ? { fontSize: "1.7rem", color: "red" }
+                  : { fontSize: "1.7rem" }
+              }
+              onClick={
+                user.isUserLoggedIn
+                  ? async () => {
+                      let wishList = [];
+                      if (!ifProductExistsInWishList) {
+                        wishList = await addToWishList(product);
+                      } else {
+                        wishList = await removeFromWishList(_id);
+                      }
+                      dispatchWishList({
+                        type: "SET_WISHLIST",
+                        payload: { value: wishList.wishlist },
+                      });
+                    }
+                  : () => navigate("/login")
```

Comment:
Suggestion: you can define function somewhere and use it here

Reply:
Done